frontend/src/repo-menu.tsx

```diff
@@ -0,0 +1,80 @@
+import { useAllPulls, useSetFilter } from './pulldasher/pulls-context';
+import { useArrayUrlState } from './use-url-state';
+import { Pull } from './pull';
+import { useConst, Button, Menu, MenuButton, MenuList, MenuDivider, MenuOptionGroup, MenuItemOption } from "@chakra-ui/react";
+import { useEffect, useCallback, useMemo } from 'react'
+
+type RepoCounts = Map<string, number>;
+
+export function RepoMenu() {
+   const pulls = useAllPulls();
+   // Default is empty array that implies show all pulls (no filtering)
+   const [selectedRepos, setSelectedRepos] = useArrayUrlState('repo', []);
+   // Nothing selected == show everything, otherwise, it'd be empty
+   const showAll = selectedRepos.length === 0;
+   // List from url may contain repos we have no pulls for
+   const urlRepos = useConst(() => new Set(selectedRepos));
+   const setPullFilter = useSetFilter();
+
+   // May include repos frmo the url for which there are no pulls
+   const allRepos = useMemo(() => {
+      // All repos of open pulls
+      const pullRepos = new Set(pulls.map((pull) => pull.getRepoName()));
+      return [...new Set([...pullRepos, ...urlRepos])]
```

Comment:
Depending, might also be worth sorting this array:
```suggestion
      return _.sortBy([...new Set([...pullRepos, ...urlRepos])])
```

or (happens to work because [`sort()` returns the array](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort)):
```suggestion
      return [...new Set([...pullRepos, ...urlRepos])].sort();
```

Reply:
Fixed in other pull.